os/arch/arm/src/imxrt/imxrt_adc.c

```diff
@@ -0,0 +1,813 @@
+/****************************************************************************
+ *
+ * Copyright 2019 NXP Semiconductors All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <string.h>
+#include <assert.h>
+#include <debug.h>
+#include <errno.h>
+#include <stdio.h>
+
+#include <tinyara/irq.h>
+#include <tinyara/wqueue.h>
+#include <tinyara/wdog.h>
+#include <tinyara/analog/adc.h>
+#include <tinyara/analog/ioctl.h>
+
+#include "imxrt_clock.h"
+#include "imxrt_adc.h"
+
+#ifdef CONFIG_ANALOG
```

Comment:
CONFIG_ANALOG for driver, not for arch. Let's revert this.

Reply:
Reverted this commit.

Reply:
reverted.